src/NServiceBus.Core.Tests/Routing/UnicastPublisherRouterTests.cs

```diff
@@ -0,0 +1,136 @@
+﻿namespace NServiceBus.Core.Tests.Routing
+{
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Extensibility;
+    using NServiceBus.Routing;
+    using NUnit.Framework;
+    using Transport;
+    using Unicast.Messages;
+    using Unicast.Subscriptions;
+    using Unicast.Subscriptions.MessageDrivenSubscriptions;
+
+    [TestFixture]
+    public class UnicastPublisherRouterTests
+    {
+        UnicastPublishRouter router;
+        MessageMetadataRegistry metadataRegistry;
+        EndpointInstances endpointInstances;
+        TransportAddresses transportAddresses;
+        FakeSubscriptionStorage subscriptionStorage;
+
+        [Test]
+        public async Task When_subscriber_does_not_define_logical_endpoint_should_send_event_to_each_address()
+        {
+            subscriptionStorage.Subscribers.Add(new Subscriber("address1", null));
+            subscriptionStorage.Subscribers.Add(new Subscriber("address2", null));
+
+            var routes = await router.Route(typeof(Event), new DistributionPolicy(), new ContextBag());
+
+            Assert.AreEqual(2, routes.Count());
+            var destinations = routes.Select(ExtractDestination).ToList();
+            Assert.Contains("address1", destinations);
+            Assert.Contains("address2", destinations);
+        }
+
+        [Test]
+        public async Task When_multiple_subscribers_for_logical_endpoints_should_route_event_to_a_single_instance_of_each_logical_endpoint()
+        {
+            var sales = "Sales";
+            var shipping = "Shipping";
+            subscriptionStorage.Subscribers.Add(new Subscriber("sales1", sales));
+            subscriptionStorage.Subscribers.Add(new Subscriber("sales2", sales));
+            subscriptionStorage.Subscribers.Add(new Subscriber("shipping1", shipping));
+            subscriptionStorage.Subscribers.Add(new Subscriber("shipping2", shipping));
+
+            var routes = (await router.Route(typeof(Event), new DistributionPolicy(), new ContextBag())).ToArray();
+
+            Assert.AreEqual(2, routes.Length);
+            var destinations = routes.Select(ExtractDestination).ToList();
+            Assert.Contains("sales1", destinations);
+            Assert.Contains("shipping1", destinations);
+        }
+
+        [Test]
+        public async Task Should_not_send_multiple_copies_of_message_to_one_physical_destination()
```

Comment:
Above you use should_route and here should_no_send (below test as well) although you still actually only get routes. Should we avoid the term send?